plugins/grid-bookmark/src/GridBookmarkWidget/model.ts

```diff
@@ -51,106 +51,93 @@ export interface IExtendedLabeledRegionModel extends ILabeledRegionModel {
   correspondingObj: ILabeledRegionModel
 }
 
+const localStorageKeyF = () =>
+  typeof window !== undefined
+    ? `bookmarks-${[window.location.host + window.location.pathname].join('-')}`
+    : 'empty'
+
 export default function f(_pluginManager: PluginManager) {
   return types
     .model('GridBookmarkModel', {
+      /**
+       * #property
+       */
       id: ElementId,
+      /**
+       * #property
+       */
       type: types.literal('GridBookmarkWidget'),
-      bookmarkedRegions: types.array(
-        types.optional(LabeledRegionModel, () =>
-          JSON.parse(
-            localStorageGetItem(
-              `bookmarks-${[
-                window.location.host + window.location.pathname,
-              ].join('-')}`,
-            ) || '[]',
-          ),
-        ),
-      ),
+      /**
+       * #property
+       * removed by postProcessSnapshot, only loaded from localStorage
+       */
+      bookmarks: types.maybe(types.array(LabeledRegionModel)),
     })
     .volatile(() => ({
       selectedBookmarks: [] as IExtendedLabeledRegionModel[],
-      localStorageKey: `bookmarks-${[
-        window.location.host + window.location.pathname,
-      ].join('-')}`,
+      selectedAssembliesPre: undefined as string[] | undefined,
     }))
     .views(self => ({
-      get sharedBookmarksModel() {
-        return SharedBookmarksModel.create({
-          sharedBookmarks: self.selectedBookmarks,
-        })
+      get bookmarkedRegions() {
+        return self.bookmarks ?? []
       },
     }))
     .views(self => ({
-      get assemblies() {
+      get bookmarkAssemblies() {
         return [...new Set(self.bookmarkedRegions.map(r => r.assemblyName))]
       },
       get validAssemblies() {
+        const { assemblyManager } = getSession(self)
         return new Set(
-          this.assemblies.filter((assembly: string) =>
-            getSession(self).assemblyNames.includes(assembly),
-          ),
+          this.bookmarkAssemblies.filter(a => assemblyManager.get(a)),
         )
       },
     }))
     .views(self => ({
       get bookmarksWithValidAssemblies() {
-        return (
-          JSON.parse(
```

Comment:
moved the clone right before creating the SharedBookmarksModel, with comment to say why the clone is needed